lib/commands/owner.js

```diff
@@ -83,8 +85,38 @@ class Owner extends BaseCommand {
     }
   }
 
+  async execWorkspaces ([action, ...args], filters) {
+    await this.setWorkspaces(filters)
+    // ls pkg or owner add/rm package
+    if ((action === 'ls' && args.length > 0) || args.length > 1) {
+      const implicitWorkspaces = this.npm.config.get('workspace', 'default')
+      if (implicitWorkspaces.length === 0) {
+        log.warn(`Ignoring specified workspace(s)`)
+      }
+      return this.exec([action, ...args])
+    }
+
+    for (const [name] of this.workspaces) {
+      switch (action) {
+        case 'ls':
+        case 'list':
+          await this.ls(name)
+          break
+        case 'add':
+          await this.changeOwners(args[0], name, 'add')
+          break
+        case 'rm':
+        case 'remove':
+          await this.changeOwners(args[0], name, 'rm')
+          break
+        default:
+          throw this.usageError()
+      }
```

Comment:
just a potential cleaner and shorter suggestion that'll be harder to mess up :-)

```suggestion
      if (action === 'ls' || action === 'list') {
          await this.ls(name)
      } else if (action === 'add') {
          await this.changeOwners(args[0], name, 'add')
      } else if (action === 'rm' || action === 'remove') {
          await this.changeOwners(args[0], name, 'rm')
      } else {
          throw this.usageError()
      }
```

Reply:
Any reason to get rid of a switch.

Reply:
Changing the non-workspace version too. `return` gives the false impression that we care about the result.